auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BasicAuthToBearerTokenFilter.java

```diff
@@ -0,0 +1,129 @@
+    private static final BaseEncoding BASE_64_ENCODING = BaseEncoding.base64Url();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public final void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        ServletRequest updatedRequest;
+        try {
+            updatedRequest = getRequestWithToken(request);
+        } catch (IllegalArgumentException e) {
+            updatedRequest = request;
+        }
+        chain.doFilter(updatedRequest, response);
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    private ServletRequest getRequestWithToken(ServletRequest request) {
+        if (request instanceof HttpServletRequest) {
+            HttpServletRequest httpRequest = (HttpServletRequest) request;
+            String rawAuthHeader = httpRequest.getHeader(HttpHeaders.AUTHORIZATION);
+            return getRequestWithTokenFromRawAuthHeader(rawAuthHeader, httpRequest);
+        } else {
+            String message = "Request is not an HttpServletRequest.";
+            log.warn(message);
+            throw new IllegalArgumentException(message);
+        }
+    }
+
+    private ServletRequest getRequestWithTokenFromRawAuthHeader(String rawAuthHeader, HttpServletRequest request) {
+        Preconditions.checkNotNull(rawAuthHeader, "null rawAuthHeader");
+        if (isBasicAuth(rawAuthHeader)) {
+            AuthHeader authHeader = base64DecodePassword(rawAuthHeader);
+            return getRequestWithTokenFromAuthHeader(authHeader, request);
+        } else {
+            String message = "Auth header is not basic auth.";
+            log.warn(message);
+            throw new IllegalArgumentException(message);
+        }
+    }
+
+    private ServletRequest getRequestWithTokenFromAuthHeader(final AuthHeader authHeader, HttpServletRequest request) {
+        return new HttpServletRequestWrapper(request) {
+            @Override
+            public String getHeader(String name) {
+                if (Objects.equals(name, HttpHeaders.AUTHORIZATION)) {
+                    return authHeader.toString();
+                } else {
+                    return super.getHeader(name);
+                }
+            }
+        };
+    }
+
+    private AuthHeader base64DecodePassword(String rawAuthHeader) {
+        String base64Credentials = rawAuthHeader.substring(BASIC_AUTH_STR.length()).trim();
+        String credentials;
+        try {
+            credentials = new String(BASE_64_ENCODING.decode(base64Credentials), StandardCharsets.UTF_8);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Could not decode credentials from auth header: " + e.getMessage());
+        }
+        String password;
+        try {
+            password = credentials.split(":", 2)[1];
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
it's not great to use exceptions for control flow. can we instead do a `checkArgument(credentials.contains(":"))`